### PR TITLE
Support dtypes in legacy new constructors.

### PR DESCRIPTION
**Pull request by gchanan**

### CODE REVIEW

**Comment by gchanan** [comment]
@colesbury I separated tensor_new (dtypes) from tensor_ctor (no dtypes).

File: torch/csrc/utils/tensor_new.cpp [hunk 1]
```diff
@@ -130,6 +132,24 @@ static Tensor new_from_sequence(const Type & type, int device, PyObject* data) {
   return new_from_data(type, device, data);
 }
 
+static void check_is_dense(const Type& type) {
+  if (type.is_sparse()) {
+    std::ostringstream oss;
+    oss << "new(..) on a dense tensor can only be called with a dense dtype, got: ";
+    oss << torch::getDtype(type)->name;
+    throw std::runtime_error(oss.str());
```
Comment (colesbury) [inline torch/csrc/utils/tensor_new.cpp]: TypeError from `Exceptions.h` might be a little better but doesn't really matter

**Review by colesbury — APPROVED** [review]
The `new()` functions are getting even harder to follow, but there's not much we can do about it I guess

**Pull request merged**